web/src/main/java/no/nav/modiapersonoversikt/service/saker/SakerServiceImpl.kt

```diff
@@ -23,100 +21,58 @@ class SakerServiceImpl : SakerService {
-        val (restSakSaker, pesysSaker) = inParallel(
-            { hentSammensatteSakerResultat(fnr) },
-            { hentPensjonSakerResultat(fnr) }
-        )
-        return slaSammenGsakPesysSaker(restSakSaker, pesysSaker)
-    }
 
-    fun hentSammensatteSakerResultat(fnr: String?): SakerService.Resultat {
-        requireFnrNotNullOrBlank(fnr)
-        val resultat = SakerService.Resultat()
-        resultat.leggTilDataFraKilde(fnr, restSakSaker)
-        resultat.leggTilDataFraKilde(fnr, arenaSaker)
-        resultat.leggTilDataFraKilde(fnr, generelleSaker)
-        resultat.leggTilDataFraKilde(fnr, oppfolgingsSaker)
-
-        leggTilFagsystemnavnOgTemanavn(
-            resultat.saker,
-            kodeverk.hentKodeverk(KodeverkConfig.FAGSYSTEM),
-            kodeverk.hentKodeverk(KodeverkConfig.ARKIVTEMA)
-        )
-
-        /**
-         * Bidragssaken må legges til etter `leggTilFagsystemnavnOgTemanavn` siden vi ikke har
-         * fagsystemkode-mapping for bidrag-hack saken
-         */
-        resultat.leggTilDataFraKilde(fnr, bidragSaker)
-
-        return resultat
+        return SakerService.Resultat()
+            .leggTilDataFraKilde(fnr, safSaker)
+            .leggTilDataFraKilde(fnr, arenaSaker)
+            .leggTilDataFraKilde(fnr, generelleSaker)
+            .leggTilDataFraKilde(fnr, oppfolgingsSaker)
+            .leggTilDataFraKilde(fnr, bidragSaker)
+            .leggTilTemaNavn(kodeverk.hentKodeverk(KodeverkConfig.ARKIVTEMA))
+            .leggTilFagsystemNavn(kodeverk.hentKodeverk(KodeverkConfig.FAGSYSTEM))
             .fjernIkkeGodkjenteSaker()
-    }
-
-    private fun SakerService.Resultat.fjernIkkeGodkjenteSaker(): SakerService.Resultat {
-        this.saker.removeIf(not(GODKJENT_FAGSAK or GODKJENT_GENERELL))
-        return this
-    }
-
-    fun hentPensjonSakerResultat(fnr: String?): SakerService.Resultat {
-        requireFnrNotNullOrBlank(fnr)
-        val resultat = SakerService.Resultat().leggTilDataFraKilde(fnr, pensjonSaker)
-        leggTilFagsystemnavnOgTemanavn(
-            resultat.saker,
-            kodeverk.hentKodeverk(KodeverkConfig.FAGSYSTEM),
-            kodeverk.hentKodeverk(KodeverkConfig.ARKIVTEMA)
-        )
-        return resultat
+            .fjernDuplikater()
     }
 
     companion object {
-        @JvmStatic
-        fun leggTilFagsystemnavnOgTemanavn(
-            saker: List<Sak>,
-            fagsystemKodeverk: EnhetligKodeverk.Kodeverk<String, String>,
-            arkivtemaKodeverk: EnhetligKodeverk.Kodeverk<String, String>,
-        ) {
-            saker.forEach {
-                it.fagsystemNavn = fagsystemKodeverk.hentVerdi(it.fagsystemKode ?: "", it.fagsystemKode ?: "")
-                it.temaNavn = arkivtemaKodeverk.hentVerdi(it.temaKode ?: "", it.temaKode ?: "")
-            }
+        private fun SakerService.Resultat.fjernIkkeGodkjenteSaker(): SakerService.Resultat {
+            this.saker.removeIf(not(::godkjentFagSak or ::godkjentGenerellSak))
```

Comment:
Spenstig syntax! Hva blir forskjellen på dette og å skrive `!` og `||` her?

Reply:
`removeIf` tar ett predikat som eneste argument, e.g `Collection<E>.removeIf(Predicate<? super E> filter)`, så det er ikke direkte mulig å bruke `!` eller `||` direkte her siden det er operasjoner som bare fungerer på boolean-verdier.

Disse to variablene vil effektivtsett være ekvivalente; 
```kotlin
private val godkjentSakPred: Predicate<Sak> = not(::godkjentFagSak or ::godkjentGenerellSak)

private val godkjentSakCode: Predicate<Sak> = Predicate { sak ->
  !(godkjentFagSak(sak) || godkjentGenerellSak(sak))
}

// Kan så kalles:
this.saker.removeIf(godkjentSakPred)

// eller
this.saker.removeIf(godkjentSakCode)
```

Så er litt smak-og-behag over hvilken løsning man går for. 

Men bakgrunnen for at det er skrevet sånn kommer fra en tid hvor filen var skrevet i Java, og da kunne man skrevet noe tilsvarende `not(GODKJENT_FAGSAK.or(GODKJENT_GENERELLSAK))`

Reply:
Endret til 
```kotlin
this.saker.removeIf {
  !godkjentFagSak(it) && !godkjentGenerellSak(it)
}
```

siden de historiske begrensningen til Java7 og tidligere ikke nå lengre er relevante, og det sannsynligvis er enklere å forstå seg på.